### PR TITLE
S3 task three review - delete button

### PR DESCRIPTION
**Pull request by EleanorRoberts**

### CODE REVIEW

File: addpokemon.php [hunk 1]
```diff
@@ -1,29 +1,44 @@
...
+            <input type="number" name="spAttack" required /></label>
+        <label for="spDefense">Sp. Defense
+            <input type="number" name="spDefense" required/></label>
+        <label for="speed">Speed
+            <input type="number" name="speed" required/></label>
+        <input type="submit" name="submit" />
+</section>
 </form>
+<footer class="footer">
+    <p>Nova Inc. Ltd</p>
```
Comment (jesshamby) [inline addpokemon.php:41]: Cute! 😍

File: collectionfunctions.php [hunk 2]
```diff
@@ -36,16 +38,26 @@ function connectToDB(String $database): PDO {
...
 }
 
-/** Collects all pokemon values from pokemon database
+/** Collects all Pokémon values from Pokémon database
  * @param PDO $database
  * @return array
  */
 function collectData(PDO $database): Array {
-    $getPokemon = $database->prepare('SELECT `stats`.`name`, `stats`.`type1`, `stats`.`type2`, `stats`.`hp`, `stats`.`attack`, `stats`.`defense`, `stats`.`spAttack`, `stats`.`spDefense`, `stats`.`speed` FROM `stats`;');
+    $getPokemon = $database->prepare("SELECT `stats`.`id`, `stats`.`name`, `stats`.`type1`, `stats`.`type2`, `stats`.`hp`, `stats`.`attack`, `stats`.`defense`, `stats`.`spAttack`, `stats`.`spDefense`, `stats`.`speed` FROM `stats` WHERE `deleted` = 0;");
```
Comment (jesshamby) [inline collectionfunctions.php]: If all the fields are in the stats table do you need to append `stats`. to each of them?

File: collectionfunctions.php [hunk 3]
```diff
@@ -81,7 +93,49 @@ function checkNew(PDO $database) {
                 header('Location: collectionchallenge.php');
```
Comment (jesshamby) [inline collectionfunctions.php:93]: Might be missing something but can this go in the if statement above? :)

File: collectionfunctions.php [hunk 4]
```diff
@@ -81,7 +93,49 @@ function checkNew(PDO $database) {
...
+            }
+        }
+    }
+}
+
+/** Soft delete Pokémon record and refresh page to show undeleted pokemon
+ * @param PDO $database
+ */
+function deletePoke(PDO $database) {
+    $deletePokemon = $database->prepare('UPDATE `stats` SET `deleted` = 1 WHERE `id` = :id LIMIT 1;');
```
Comment (jesshamby) [inline collectionfunctions.php]: Not sure if you need the limit if you have the WHERE as your id as you should never have more than one row with the same id? :)

**Review by jesshamby — COMMENTED** [review]
Looks awesome! Just a few small comments, although a lot of them are more like questions :)

File: collectionfunctions.php (hunk 4 above)
Comment (EleanorRoberts) [reply collectionfunctions.php]: I put it in as an extra check, but I can remove it :)

File: collectionfunctions.php (hunk 3 above)
Comment (EleanorRoberts) [reply collectionfunctions.php:93]: This is where I actually call the execute, but I need to do the checks in the if statement above before I do 😇

File: collectionfunctions.php (hunk 2 above)
Comment (EleanorRoberts) [reply collectionfunctions.php]: Probably not 🧐 I think I put them in there as I thought it might be best practice in case it's swapped for a join but idk

**Pull request merged**